src/main/java/mate/academy/springboot/datajpa/controller/ProductController.java

```diff
@@ -0,0 +1,79 @@
+package mate.academy.springboot.datajpa.controller;
+
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import mate.academy.springboot.datajpa.dto.request.ProductRequestDto;
+import mate.academy.springboot.datajpa.dto.response.ProductResponseDto;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.services.CategoryService;
+import mate.academy.springboot.datajpa.services.ProductService;
+import mate.academy.springboot.datajpa.services.mapper.ProductMapper;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/products")
+@RequiredArgsConstructor
+public class ProductController {
+    private final ProductService productService;
+    private final ProductMapper productMapper;
+    private final CategoryService categoryService;
+
+    @PostMapping
+    public ProductResponseDto create(@RequestBody ProductRequestDto productRequestDto) {
+        return productMapper.toDto(
+                productService.create(
+                        productMapper.toModel(productRequestDto)));
+    }
+
+    @GetMapping("/{id}")
+    public ProductResponseDto getById(@PathVariable Long id) {
+        return productMapper.toDto(productService.getByID(id));
+    }
+
+    @DeleteMapping("/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    public void delete(@PathVariable Long id) {
+        productService.deleteById(id);
+    }
+
+    @PutMapping("/{id}")
+    public ProductResponseDto update(@PathVariable Long id,
+                                     @RequestBody ProductRequestDto productRequestDto) {
+        Product product = productService.getByID(id);
+        product.setTitle(productRequestDto.getTitle());
+        product.setCategory(categoryService.getById(productRequestDto.getCategoryId()));
+        product.setPrice(productRequestDto.getPrice());
```

Comment:
first of all you don't need to user getById, 
all this logic should be moved to mapper